strawberry/channels/handlers/http_handler.py

```diff
@@ -0,0 +1,238 @@
+        else:
+            await self.send_response(
+                result.status,
+                result.response,
+                headers=[(b"Content-Type", result.content_type.encode())],
+            )
+
+    async def get(self, body: bytes) -> Result:
+        if self.should_render_graphiql():
+            return await self.render_graphiql(body)
+        elif self.scope.get("query_string"):
+            params = parse_query_params(
+                {
+                    k: v[0]
+                    for k, v in parse_qs(self.scope["query_string"].decode()).items()
+                }
+            )
+            if "query" not in params:
+                raise ExecutionError("No GraphQL query found in the request")
+
+            result = await self.execute(parse_request_data(params))
+            return Result(response=json.dumps(result).encode())
+        else:
+            raise MethodNotAllowed()
+
+    async def post(self, body: bytes) -> Result:
+        request_data = await self.parse_body(body)
+        result = await self.execute(request_data)
+        return Result(response=json.dumps(result).encode())
+
+    async def parse_body(self, body: bytes) -> GraphQLRequestData:
+        if self.headers.get("content-type", "").startswith("multipart/form-data"):
+            return await self.parse_multipart_body(body)
+
+        try:
+            data = json.loads(body)
+        except json.JSONDecodeError as e:
+            raise ExecutionError("Unable to parse request body as JSON") from e
+
+        try:
+            return parse_request_data(data)
+        except MissingQueryError as e:
+            raise ExecutionError("No GraphQL query found in the request") from e
+
+    async def parse_multipart_body(self, body: bytes) -> GraphQLRequestData:
+        raise ExecutionError("Unable to parse the multipart body")
+
+    async def execute(self, request_data: GraphQLRequestData):
+        context = await self.get_context()
+        root_value = await self.get_root_value()
+
+        method = self.scope["method"]
+        allowed_operation_types = OperationType.from_http(method)
+        if not self.allow_queries_via_get and method == "GET":
+            allowed_operation_types = allowed_operation_types - {OperationType.QUERY}
+
+        result = await self.schema.execute(
+            query=request_data.query,
+            root_value=root_value,
+            variable_values=request_data.variables,
+            context_value=context,
+            operation_name=request_data.operation_name,
+            allowed_operation_types=allowed_operation_types,
+        )
+        return await self.process_result(result)
+
+    async def process_result(self, result: ExecutionResult) -> GraphQLHTTPResponse:
+        return process_result(result)
+
+    async def render_graphiql(self, body):
+        html_string = self.graphiql_html_file_path.read_text()
+        html_string = html_string.replace(
+            "{{ SUBSCRIPTION_ENABLED }}",
```

Comment:
should we assume true for this, since we'll be using channels for subscriptions?

Reply:
I thought the same, but because of the http consumer you can actually have a graphql integration using only queries and mutations over websockets without the subscriptions. But I think it is fair to assume that the majority of people using this is for subscriptions.

If you prefer, I can remove the option and always enable it

Reply:
ah, good point, let's leave like so for now 😊